src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,45 @@
+import {
+    Component,
+    EventEmitter,
+    Input,
+    Output,
+    TemplateRef,
+    ViewChild,
+    ViewContainerRef,
+} from '@angular/core';
+
+@Component({
+    selector: 'app-popup-host',
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('viewport', {read: ViewContainerRef, static: true})
+    private readonly viewport: ViewContainerRef | undefined;
+
+    @Input() set template(templateRef: TemplateRef<unknown> | null) {
+        if (templateRef) {
+            this.insertModalTemplate(templateRef);
+        } else {
+            this.initViewport(true);
+        }
```

Comment:
Чуть чуть попродаю best practics)

Не рекомендую к использованию конструкцию `if/else`, особенно в комбинации нескольких `else` (`if / else if / else`). Вместо этого рекомендую использовать ранний `return`, он более гибки позволяет делать промежуточны вычисления, и, соответсвенно, писать более чистый код)
```suggestion
        if (templateRef) {
            this.insertModalTemplate(templateRef);
            
            return;
        }
        
        this.initViewport(true);
```

Reply:
Егор, спасибо - отличный совет. По началу только смутило про ранний return в сеттере )